stylo/shape/shape.py

```diff
@@ -78,15 +82,43 @@ def draw(self):
         pass
 
 
-def composite_shape_factory(op, name):
+class InvertedShape(Shape):
+    """A shape that has been inverted."""
+
+    def __init__(self, shape):
+        self.shape = shape
+
+    def __repr__(self):
+        return "NOT [ " + repr(self.shape) + " ]"
+
+    def _render_domain(self, *args):
+        mask = self.shape._render_domain(*args)
+        return np.logical_not(mask)
+
+    def _render_args(self, **kwargs):
+        mask = self.shape._render_args(**kwargs)
+        return np.logical_not(mask)
+
+    def draw(self):
+        pass
+
+    @property
+    def parameters(self):
+        return self.shape.parameters
+
+
+def composite_shape_factory(op, name, op_name):
     """A factory function that returns class definitions for composite shapes
-    constructed using logical operations such as AND, OR, etc.
+    constructed using binary logical operations such as AND, OR, etc.
 
     :param op: The function that implements the operation in question
     :param name: The name to give the composite class.
+    :param op_name: The name of the logical operation, this will be used in the objcet's
```

Comment:
typo in `objcet's`

Reply:
I should really be using a spell checker... :smile: